src/main/java/pl/javastart/task/TournamentStats.java

```diff
@@ -1,12 +1,57 @@
 package pl.javastart.task;
 
-import java.util.Scanner;
+import java.util.*;
 
 public class TournamentStats {
 
+    private static final String STOP = "STOP";
+    private static final int FIRST_NAME = 1;
+    private static final int LAST_NAME = 2;
+    private static final int RESULT = 3;
+    private static final int DESCENDING = 2;
+
     void run(Scanner scanner) {
-        // tutaj dodaj swoje rozwiązanie
-        // użyj przekazanego scannera do wczytywania wartości
+        String result;
+        List<Participant> participants = new ArrayList<>();
+        do {
+            result = getResult(scanner);
+            addParticipantResult(result, participants);
+        } while (!result.equals(STOP));
+        sortByParameter(scanner, participants);
+        sortAscendingOrDescending(scanner, participants);
```

Comment:
Sortujesz kolekcję dwukrotnie, spróbuj zapisać to tak, żeby metoda sort() była wywołana tylko raz